### PR TITLE
feat(totp): initial totp implementation 

### LINKED ISSUE #5863: TOTP Setup flow initial UX
To help with planning here are the initial requirements for this flow. Based on [doc](https://docs.google.com/document/d/1m4D_7DZAfXt4e-bEbfI-5s_x2XFO7TEQo1a_SQcj1O8/edit#heading=h.yzz637baqr7a).

- There is a new "settings" page section or sub-screen, where users can opt-in to using TOTP as a second factor for authentication on their account.
- The MFA settings are not visible by default. 
- The MFA settings are visible to users who specify a feature-flag in the URL 
- The MFA settings are visible to users who have opted-in to MFA on their account (optional)
- Users who visit the MFA settings page can enroll a TOTP device or app (e.g. Google Authenticator) by scanning a QR code.
- After enrollment, the user is presented with backup codes to store in case they lose their MFA device. 
- After enrollment, the user must submit a TOTP code to verify correct setup. 

This version will contain rough user interfaces based on this [doc](https://www.lucidchart.com/documents/view/833ea3ae-65bc-4b0a-94cb-3290ec27ae85). These requirements and designs may change based on user and security reviews.

### PR DESCRIPTION
**Pull request by vbudhram**
This PR is the same as https://github.com/mozilla/fxa-content-server/pull/5898 (with review fixes), but targeted against train 107.

Only minor update is that the feature is behind a query param, `?showTwoStepAuthentication` since the panel name was updated.

@mozilla/fxa-devs @vladikoff r?

Fixes #5863 
Fixes #5898

### CODE REVIEW

File: app/scripts/templates/sign_in_totp_code.mustache [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    <div class="graphic graphic-two-factor-auth"></div>
+
+    <p class="verification-totp-message">
+    {{#t}}Open your authentication app and enter the security code it provides.{{/t}}
+    </p>
+
+    <form novalidate>
+      <div class="input-row token-code-row">
+        <input type="text" class="tooltip-below totp-code" placeholder="{{#t}}Enter 6-digit code{{/t}}" required autofocus />
```
Comment (vladikoff) [inline app/scripts/templates/sign_in_totp_code.mustache]: can this be `type` `number` ?

File: app/scripts/views/settings/two_step_authentication.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  },
+
+  _hideStatus() {
+    this.$('.security-list').addClass('hidden');
+    this.$('.totp-refresh').addClass('hidden');
+    this.$('.totp-confirm-code').removeClass('hidden');
+  },
+
+  _showCode() {
+    $('.manual-code').removeClass('hidden');
```
Comment (vladikoff) [inline app/scripts/views/settings/two_step_authentication.js]: can this use `this.$`?

File: app/scripts/views/settings/two_step_authentication.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    $('.show-code-link').addClass('hidden');
```
Comment (vladikoff) [inline app/scripts/views/settings/two_step_authentication.js]: can this use `this.$`?

File: app/scripts/views/settings/two_step_authentication.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  cancel() {
+    return this.render()
+      .then(() => this.navigate('/settings'));
+  },
+
+  createToken() {
+    const account = this.getSignedInAccount();
+    return account.createTotpToken()
+      .then(result => {
+        $('.qr-image').attr('src', result.qrCodeUrl);
```
Comment (vladikoff) [inline app/scripts/views/settings/two_step_authentication.js]: can this use `this.$`?

File: app/scripts/views/settings/two_step_authentication.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        $('.code').html(this._getFormattedCode(result.secret));
```
Comment (vladikoff) [inline app/scripts/views/settings/two_step_authentication.js]: can this use `this.$`?

File: app/scripts/views/settings/two_step_authentication.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+    const account = this.getSignedInAccount();
+    return account.deleteTotpToken()
+      .then(() => {
+        this.displaySuccess(t('Two-step authentication removed'), {
+          closePanel: true
+        });
+        return this.render()
+          .then(() => {
+            this.navigate('/settings');
+          });
```
Comment (vladikoff) [inline app/scripts/views/settings/two_step_authentication.js]: can this logic above not pyramid stack but continue flat with promises? 

I see a `render` then a navigate that also renders? Could we use something like this to signal the message to show https://github.com/mozilla/fxa-content-server/blob/08e931e8482e71dc8359d0d97fe3b779c89844d1/app/scripts/views/mixins/connect-another-device-mixin.js#L71 ?

File: app/scripts/views/sign_in_totp_code.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
```
Comment (vladikoff) [inline app/scripts/views/sign_in_totp_code.js]: Try removing `define(function (require, exports, module) {` in this file as well, should work!

File: app/tests/spec/views/settings/two_step_authentication.js [hunk 8]
```diff
@@ -0,0 +1,195 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
```
Comment (vladikoff) [inline app/tests/spec/views/settings/two_step_authentication.js]: try removing `define(function (require, exports, module) {`

File: app/tests/spec/views/settings/two_step_authentication.js [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+  'use strict';
```
Comment (vladikoff) [inline app/tests/spec/views/settings/two_step_authentication.js]: also won't need this i think, webpack should add it

File: app/tests/spec/views/sign_in_totp_code.js [hunk 10]
```diff
@@ -0,0 +1,188 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
```
Comment (vladikoff) [inline app/tests/spec/views/sign_in_totp_code.js]: `define(function (require, exports, module) {`

**Comment by vbudhram** [comment]
@vladikoff Thank you!

**Pull request merged**